providers/go-feature-flag/src/test/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProviderTest.java

```diff
@@ -180,41 +176,50 @@ public void initialize(EvaluationContext evaluationContext) throws Exception {
         assertThrows(ProviderNotReadyError.class, ()-> g.getBooleanEvaluation("bool_targeting_match", false, this.evaluationContext));
 
         String providerName = "shouldReturnNotReadyIfNotInitialized";
-        OpenFeatureAPI.getInstance().setProvider(providerName, g);
+        OpenFeatureAPI.getInstance().setProviderAndWait(providerName, g);
         assertThat(OpenFeatureAPI.getInstance().getProvider(providerName).getState()).isEqualTo(ProviderState.NOT_READY);
         Client client = OpenFeatureAPI.getInstance().getClient(providerName);
         FlagEvaluationDetails<Boolean> booleanFlagEvaluationDetails = client.getBooleanDetails("return_error_when_not_initialized", false, new ImmutableContext("targetingKey"));
         assertEquals(ErrorCode.PROVIDER_NOT_READY, booleanFlagEvaluationDetails.getErrorCode());
         assertEquals(Boolean.FALSE, booleanFlagEvaluationDetails.getValue());
+        OpenFeatureAPI.getInstance().shutdown();
```

Comment:
No need on each method since it is added in _afterEach()_ ?
same in other methods.